### ISSUE TITLE
 Exception AttributeError in training a HMM (code was working with yahmm) (referencing #82) 

### ISSUE THREAD

**Issue by narendramukherjee** [jmschrei/pomegranate#91]
This is a continuation of issue #82 

I forgot to mention this the last time - I had tested the code with the already implemented DiscreteDistribution (for a different dataset) and the same error had popped up. I tried the PoissonDistribution today, and the error came up again - I am pasting the code here again (I have commented out my definition of the PoissonDistribution, and am using the version you implemented last time).

And here's the error:
...
Exception AttributeError: 'exit' in 'pomegranate.hmm.HiddenMarkovModel._train_loop' ignored
Total Training Improvement: 0.0
...

The code:
[test_pomegranate.txt](https://github.com/jmschrei/pomegranate/files/160105/test_pomegranate.txt)

**Comment by jmschrei** [comment]
Hey, thanks for the update. When I run that code, I get the following:

```
jmschr@ubuntu:~/Desktop/pomegranate$ python test_poisson.py
Training improvement: 14107.7097227
Training improvement: 3645.61835847
Training improvement: 401.01703992
Training improvement: 5.39640137579
Training improvement: 0.000292382926091
Training improvement: 2.59205990005e-10
Total Training Improvement: 18159.7418148
```

Can you tell me more about your system? Did you pull the latest github code?

**Comment by narendramukherjee** [comment]
Yes, I pulled the latest code (I downloaded the zip version from github and installed it by running setup.py install). Here's my python terminal:
...
narendra@spokeyspikes:~$ ipython
Python 2.7.11 |Anaconda 2.5.0 (64-bit)| (default, Dec  6 2015, 18:08:32) 
Type "copyright", "credits" or "license" for more information.
...
I am running Ubuntu 14.04.4 LTS.

Thanks for your help all along!

**Comment by jmschrei** [comment]
I'm really having trouble reproducing your error. Are you positive that you properly installed the new version?

**Comment by narendramukherjee** [comment]
I am positive that I installed the new version because the PoissonDistribution class worked out of the box without me defining it. I had the same error when I used the pre-defined DiscreteDistribution class too - it might however be something weird with stuff on my computer. Let me try this on a different computer and then I'll update on how things look.

**Comment by jmschrei** [comment]
Any update on the different computer?

**Comment by narendramukherjee** [comment]
I re-installed Ubuntu afresh on my computer (and moved up to 15.10) - then I re-installed pomegranate on Anaconda. The same issue comes up as last time - PoissonDistribution works (I can assign a PoissonDistribution object without problems), but I get the same AttributeError in _train_loop :( Since I am working with a fresh system now and things don't seem to be working on Anaconda, can you list the dependencies I should have to get pomegranate working on my system Python? Then I can try installing these dependencies with pip on my system Python, and see if that makes a difference?

**Comment by narendramukherjee** [comment]
Update: I got this working - basically, I was trying to install using setup.py install previously. That doesn't seem to install joblib (which is needed) however. This comes up as an error when I try to import pomegranate later - at that point, I used conda to install joblib. Conda, however, installs joblib 0.8 which seems to be the problem.

Fix: I instead did pip install pomegranate (after I had pulled the code into a folder called pomegranate) - pip pulls out joblib 0.9 and installs it before installing pomegranate. That seems to fix everything, and my code works.

Thanks for your help Jacob, I am closing this issue now...
